clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -272,28 +272,28 @@ class GCEUtil {
   static def queryHealthCheck(String projectName,
                               String account,
                               String healthCheckName,
+                              GoogleHealthCheck.HealthCheckKind healthCheckKind,
                               Compute compute,
                               Cache cacheView,
                               Task task,
                               String phase,
                               GoogleExecutorTraits executor) {
-    task.updateStatus phase, "Looking up http(s) health check $healthCheckName..."
-
-    def httpHealthCheckIdentifiers = cacheView.filterIdentifiers(HTTP_HEALTH_CHECKS.ns, Keys.getHttpHealthCheckKey(account, healthCheckName))
-    def results = cacheView.getAll(HTTP_HEALTH_CHECKS.ns, httpHealthCheckIdentifiers, RelationshipCacheFilter.none())
+    task.updateStatus phase, "Looking up health check $healthCheckName..."
 
-    if (results[0]?.attributes?.httpHealthCheck) {
-      return results[0]?.attributes?.httpHealthCheck
-    } else {
-      try {
-        // TODO(duftler): Update this to use the cache instead of a live call once we are caching https health checks.
-        return executor.timeExecute(
-          compute.httpsHealthChecks().get(projectName, healthCheckName),
-          "compute.httpsHealthChecks.get",
-          executor.TAG_SCOPE, executor.SCOPE_GLOBAL)
-      } catch (GoogleJsonResponseException | SocketTimeoutException | SocketException _) {
-        updateStatusAndThrowNotFoundException("Http(s) health check $healthCheckName not found.", task, phase)
-      }
+    switch (healthCheckKind) {
+      case GoogleHealthCheck.HealthCheckKind.healthCheck:
+        return queryNestedHealthCheck(projectName, account, healthCheckName, compute, cacheView, task, phase, executor)
+      case GoogleHealthCheck.HealthCheckKind.httpHealthCheck:
+        return queryLegacyHttpHealthCheck(account, healthCheckName, cacheView, task, phase)
+      case GoogleHealthCheck.HealthCheckKind.httpsHealthCheck:
+        return queryLegacyHttpsHealthCheck(projectName, healthCheckName, compute, task, phase, executor)
+      default:
+        // Note: Cache queries for these health checks must occur in this order since queryLegacyHttpsHealthCheck() will make a live
+        // call that fails on a missing health check.
+        // todo(mneterval): return null instead of querying for each type once all health check payloads include `healthCheckKind`
```

Comment:
Are the health checks missing `healthCheckKind` in incoming operations e.g. from the UI?

Reply:
Yep, SGTM.